### ISSUE TITLE
callback returning null when being processed by a promise inside

### ISSUE THREAD

**Issue by danova123** [apache/cordova-plugin-inappbrowser#414]
Here is the scenario.
I have a form input and a submit button which redirects to a different page onclick. I have to read the input and store it on localstorage from the form before redirection. I tried doing LoadStart/LoadStop with reading the DOM directly, it didn't work because the page was getting changed and then the loadstart and loadstop events couldn't reference the previous DOM. So, i put it on the button click event as a promise.
This happens only on the Android app. iOS catches the loadstart event correctly.

**Call 1** ( Direct Return without click which works correctly )
```
inAppBrowserRef.executeScript({
                code: "getData();" +
                         "function getData(){" + 
                         "var name = "John";" + // Here, i will read the input
                         "alert(name);" +  //alerts John
                         "return name;",
              function(results){
                console.log(results); //THIS WORKS, it returns John
  }); 
``` 
 
**Call 2:** ( Indirect Return On Button Click which doesn't work )
```
inAppBrowserRef.executeScript({
                code: "getData(); function getData() { return new Promise(function(resolve) { document.getElementById('submitButton').click = function() { alert('John'); resolve('John'); } }); }"
              }, function(results){
                console.log("Loadstop Callback", results); // THIS RETURNS BLANK OBJECT!!!!!!!!!!
              });
```

Do i have to use BeforeLoad event or how to achieve this.

In a nutshell, i want to pass a field from the webview to my main view and vice versa

Thanks

**Comment by danova123** [comment]
@jcesarmobile sorry to tag you because our Android app is failing to work. Is this a plugin issue or is it a coding issue. Because it was working earlier.

**Comment by jcesarmobile** [comment]
🤷🏻‍♂️

**Comment by danova123** [comment]
Found the issue. In the case of URL Redirection, the ExecuteScript of loadstop of the first page was delayed so it was not fired until the loadstop of the second page(redirected page) where it did not find the element. 
"BeforeLoad" event came to the rescue. Thanks for adding this amazing feature
Solution - Solved using beforeload event. Closing the issue.
